fv3core/fv3core/stencils/d_sw.py

```diff
@@ -226,17 +223,22 @@ def kinetic_energy_update_part_1(
         dy (in):
         dya (???):
         rdy (in):
-        ub_contra (out):
-        vb_contra (out):
-        advected_u (out):
-        advected_v (out):
+        dt_kinetic_energy_on_cell_corners (out): kinetic energy on cell corners,
+            as defined in FV3 documentation by equation 6.3, multiplied by dt
+        dt: timestep
     """
     with computation(PARALLEL), interval(...):
         ub_contra, vb_contra = interpolate_uc_vc_to_cell_corners(
             uc, vc, cosa, rsina, uc_contra, vc_contra
         )
         advected_v = advect_v_along_y(v, vb_contra, rdy=rdy, dy=dy, dya=dya, dt=dt)
         advected_u = advect_u_along_x(u, ub_contra, rdx=rdx, dx=dx, dxa=dxa, dt=dt)
+        dt_kinetic_energy_on_cell_corners = (
+            0.5 * dt * (ub_contra * advected_u + vb_contra * advected_v)
+        )
+        dt_kinetic_energy_on_cell_corners = all_corners_ke(
```

Comment:
I think this merge should be illegal because you are using `ub_contra` and `vb_contra` in regions after writing them in the first part of the stencil.

Unclear why this was not caught automatically - maybe @jdahm  has more insight?

Reply:
scratch all of that, we're only writing `ub_contra`, `vb_contra` but are reading `uc_contra`, `vc_contra`